### PR TITLE
Merge xcconfig files

### PR DESCRIPTION
**Pull request by jasssonpet**
https://github.com/NativeScript/ios-runtime/pull/363

### CODE REVIEW

**Comment by jasssonpet** [comment]
Some relative paths in the xcconfig files would get messed up now, with the merge strategy..

File: lib/services/ios-project-service.ts [hunk 1]
```diff
@@ -624,9 +607,38 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
+			this.$childProcess.exec(`ruby -e "${mergeScript}"`).wait();
+		}).future<void>()();
+	}
+
+	private regeneratePluginsXcconfigFile(): IFuture<void> {
+		return (() => {
+			this.$fs.deleteFile(this.pluginsDebugXcconfigFilePath).wait();
+			this.$fs.deleteFile(this.pluginsReleaseXcconfigFilePath).wait();
+
+			let allPlugins: IPluginData[] = (<IPluginsService>this.$injector.resolve("pluginsService")).getAllInstalledPlugins().wait();
```
Comment (Fatme) [inline lib/services/ios-project-service.ts]: Can we use constructor injection?

Comment (jasssonpet) [inline lib/services/ios-project-service.ts]: There were some circular dependencies.

**Comment by Fatme** [comment]
:+1:

**Pull request merged**